include/stdx/tuple_algorithms.hpp

```diff
@@ -156,21 +159,23 @@ constexpr auto contains_type =
     decltype(detail::contains_type<T>(std::declval<Tuple>()))::value;
 
 template <template <typename> typename Proj = std::type_identity_t,
-          typename... Ts>
-[[nodiscard]] constexpr auto sort(stdx::tuple<Ts...> t) {
+          typename Tuple>
+[[nodiscard]] constexpr auto sort(Tuple &&t) {
+    using T = stdx::remove_cvref_t<Tuple>;
     using P = std::pair<std::string_view, std::size_t>;
     constexpr auto indices = []<std::size_t... Is>(std::index_sequence<Is...>) {
         auto a = std::array<P, sizeof...(Is)>{
-            P{stdx::type_as_string<Proj<Ts>>(), Is}...};
+            P{stdx::type_as_string<Proj<tuple_element_t<Is, T>>>(), Is}...};
```

Comment:
praise: I absolutely love that we are still sorting the types by string name...so fast and awesome.